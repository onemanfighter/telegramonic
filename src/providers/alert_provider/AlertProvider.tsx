import { AlertComponent } from '@components';
import { appStore } from '@store';
import { AlertSelector, useShallow } from '@selectors';

import { AlertProviderProps } from './types';

const AlertProvider = ({ children }: AlertProviderProps) => {
  const { alert, visible } = appStore(useShallow(AlertSelector));

  return (
    <>
      {visible && <AlertComponent {...alert} />}
      {children}
    </>
  );
};

export default AlertProvider;
