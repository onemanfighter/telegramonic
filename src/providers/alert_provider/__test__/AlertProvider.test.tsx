import { act, render, renderHook, screen } from '@testing-library/react';
import AlertProvider from '../AlertProvider';
import { AlertSelector, useShallow } from '@selectors';
import { AlertComponentState, appStore } from '@store';

jest.useFakeTimers();

describe('AlertProvider', () => {
  const alertComponentState: AlertComponentState = {
    title: 'Title',
    description: 'Description',
    showIcon: true,
    status: 'success',
    variant: 'solid',
  };

  it('should render children', () => {
    const { container } = render(
      <AlertProvider>
        <div>child</div>
      </AlertProvider>,
    );
    expect(container).toMatchSnapshot();
    expect(screen.getByText('child')).toBeDefined();
  });

  it('should open alert component on setting visible state to true', () => {
    const { result } = renderHook(() => appStore(useShallow(AlertSelector)));
    const { container } = render(
      <AlertProvider>
        <div>child</div>
      </AlertProvider>,
    );

    act(() => {
      result.current.setAlertStateData(alertComponentState);
      result.current.setAlertVisible(true);
    });

    expect(container).toMatchSnapshot();
    expect(screen.getByText('child')).toBeDefined();
    expect(screen.getByText('Title')).toBeDefined();
    expect(screen.getByText('Description')).toBeDefined();
  });

  it('should show alert for the given time using showAlertForTime', () => {
    const { result } = renderHook(() => appStore(useShallow(AlertSelector)));
    const { container } = render(
      <AlertProvider>
        <div>child</div>
      </AlertProvider>,
    );

    act(() => {
      result.current.showAlertForTime(alertComponentState, 2000);
    });

    expect(container).toMatchSnapshot();
    expect(screen.getByText('child')).toBeDefined();
    expect(screen.getByText('Title')).toBeDefined();
    expect(screen.getByText('Description')).toBeDefined();

    act(() => {
      jest.runAllTimers();
    });

    expect(container).toMatchSnapshot();
    expect(screen.queryByText('Title')).toBeNull();
    expect(screen.queryByText('Description')).toBeNull();
  });
});
