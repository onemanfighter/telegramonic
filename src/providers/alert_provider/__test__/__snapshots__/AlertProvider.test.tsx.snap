// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertProvider should open alert component on setting visible state to true 1`] = `
<div>
  <div
    class="chakra-alert css-1hwd9xd"
    data-status="success"
    role="alert"
  >
    <span
      class="chakra-alert__icon css-1c6vscd"
      data-status="success"
    >
      <svg
        class="chakra-icon css-qk6lof"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12,0A12,12,0,1,0,24,12,12.014,12.014,0,0,0,12,0Zm6.927,8.2-6.845,9.289a1.011,1.011,0,0,1-1.43.188L5.764,13.769a1,1,0,1,1,1.25-1.562l4.076,3.261,6.227-8.451A1,1,0,1,1,18.927,8.2Z"
          fill="currentColor"
        />
      </svg>
    </span>
    <div
      class="chakra-alert__title css-0"
      data-status="success"
    >
      Title
    </div>
    <div
      class="chakra-alert__desc css-6n7j50"
      data-status="success"
    >
      Description
    </div>
  </div>
  <div>
    child
  </div>
</div>
`;

exports[`AlertProvider should render children 1`] = `
<div>
  <div>
    child
  </div>
</div>
`;

exports[`AlertProvider should show alert for the given time using showAlertForTime 1`] = `
<div>
  <div
    class="chakra-alert css-1hwd9xd"
    data-status="success"
    role="alert"
  >
    <span
      class="chakra-alert__icon css-1c6vscd"
      data-status="success"
    >
      <svg
        class="chakra-icon css-qk6lof"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12,0A12,12,0,1,0,24,12,12.014,12.014,0,0,0,12,0Zm6.927,8.2-6.845,9.289a1.011,1.011,0,0,1-1.43.188L5.764,13.769a1,1,0,1,1,1.25-1.562l4.076,3.261,6.227-8.451A1,1,0,1,1,18.927,8.2Z"
          fill="currentColor"
        />
      </svg>
    </span>
    <div
      class="chakra-alert__title css-0"
      data-status="success"
    >
      Title
    </div>
    <div
      class="chakra-alert__desc css-6n7j50"
      data-status="success"
    >
      Description
    </div>
  </div>
  <div>
    child
  </div>
</div>
`;

exports[`AlertProvider should show alert for the given time using showAlertForTime 2`] = `
<div>
  <div>
    child
  </div>
</div>
`;
